### PR TITLE
Test almost everything

### PR DESCRIPTION
**Pull request by sandlerben**
Added osWrapper struct to visualization.go and main.go to wrap Go os and exec functions. This enables much more test mocking.

### CODE REVIEW

**Comment by sandlerben** [comment]
Increased coverage 9% :sunglasses:

**Comment by sandlerben** [comment]
r @sectioneight

File: visualization/visualization_test.go [hunk 1]
```diff
@@ -97,6 +97,42 @@ func TestGenerateFlameGraphExecError(t *testing.T) {
 	mockExecutor.AssertExpectations(t)
 }
 
+func TestRunPerlScriptDoesExist(t *testing.T) {
+	mockOSWrapper := new(mockOSWrapper)
+	executor := defaultExecutor{
+		osWrapper: mockOSWrapper,
+	}
+	cwd, _ := os.Getwd()
```
Comment (sectioneight) [inline visualization/visualization_test.go]: may want to check errors here

File: main.go [hunk 2]
```diff
@@ -59,11 +59,19 @@ type validator interface {
...
+
+type defaultOSWrapper struct{}
+
 type pprofer interface {
 	runPprofCommand(args ...string) ([]byte, error)
 }
 
-type defaultPprofer struct{}
+type defaultPprofer struct {
+	osWrapper osWrapper
```
Comment (sectioneight) [inline main.go]: i think this line can just be:

```
  osWrapper
```

**Comment by sectioneight** [comment]
looks good

File: visualization/visualization_test.go (hunk 1 above)
Comment (sandlerben) [inline visualization/visualization_test.go]: If `os.Getwd()` fails then runPerlScript is going to propagate the error when it calls `os.Getwd()`. Hence, this test will fail even if I don't check errors here.

Comment (sectioneight) [inline visualization/visualization_test.go]: sure, i just mean it's bad form to discard errors, and you would want to fail fast here to indicate that something is messed up with your test environment

Comment (sandlerben) [inline visualization/visualization_test.go]: makes sense

**Pull request merged**